src/librustdoc/passes/mod.rs

```diff
@@ -348,3 +360,50 @@ impl fold::DocFolder for ImportStripper {
         }
     }
 }
+
+pub fn look_for_tests<'a, 'tcx: 'a, 'rcx: 'a, 'cstore: 'rcx>(
+    cx: &'a DocContext<'a, 'tcx, 'rcx, 'cstore>,
+    dox: &str,
+    item: &Item,
+    check_missing_code: bool,
+) {
+    if (item.is_mod() && cx.tcx.hir.as_local_node_id(item.def_id).is_none()) ||
+       cx.as_local_node_id(item.def_id).is_none() {
```

Comment:
Can't this just be reduced to a single `cx.as_local_node_id(item.def_id).is_none()` ? That defers to `cx.tcx.hir` anyway, without having to to the extra module check.